pkg/generate/code/set_sdk.go

```diff
@@ -718,6 +722,139 @@ func SetSDKSetAttributes(
+// with the *single* VpcId value to ensure the returned array from the API call
+// consists only of the desired Vpc.
+func setSDKReadMany(
+	cfg *ackgenconfig.Config,
+	r *model.CRD,
+	op *awssdkmodel.Operation,
+	sourceVarName string,
+	targetVarName string,
+	indentLevel int,
+) string {
+	inputShape := op.InputRef.Shape
+	if inputShape == nil {
+		return ""
+	}
+
+	out := "\n"
+	indent := strings.Repeat("\t", indentLevel)
+
+	resVarPath := ""
+	pluralize := pluralize.NewClient()
+	opConfig, override := cfg.OverrideValues(op.Name)
+	shapeIdentifiers := FindIdentifiersInShape(r, inputShape)
+	var err error
+	for memberIndex, memberName := range inputShape.MemberNames() {
+		if override {
+			value, ok := opConfig[memberName]
+			memberShapeRef, _ := inputShape.MemberRefs[memberName]
+			memberShape := memberShapeRef.Shape
+			if ok {
+				switch memberShape.Type {
+				case "boolean", "integer":
+				case "string":
+					value = "\"" + value + "\""
+				default:
+					panic("Member type not handled")
+				}
+
+				out += fmt.Sprintf("%s%s.Set%s(%s)\n", indent, targetVarName, memberName, value)
+				continue
+			}
+		}
+
+		// Field renames are handled in getSanitizedMemberPath
+		resVarPath, err = getSanitizedMemberPath(memberName, r, op, sourceVarName)
+		if err != nil {
+			// if it's an identifier field check for singular/plural
+			if util.InStrings(memberName, shapeIdentifiers) {
+				var flipped string
+				if pluralize.IsPlural(memberName) {
+					flipped = pluralize.Singular(memberName)
+				} else {
+					flipped = pluralize.Plural(memberName)
+				}
+				// If there are multiple identifiers, then prioritize the
+				// 'Id' identifier.
+				if resVarPath == "" || (!strings.HasSuffix(resVarPath, "Id") ||
+					!strings.HasSuffix(resVarPath, "Ids")) {
+					resVarPath, err = getSanitizedMemberPath(flipped, r, op, sourceVarName)
+				}
+			} else {
+				// TODO(jaypipes): check generator config for exceptions?
+				continue
+			}
+		}
+
+		memberShapeRef, _ := inputShape.MemberRefs[memberName]
+		memberShape := memberShapeRef.Shape
+		out += fmt.Sprintf(
+			"%sif %s != nil {\n", indent, resVarPath,
+		)
+
+		switch memberShape.Type {
+		case "list":
+			// Expecting slice of identifiers
+			{
+				memberVarName := fmt.Sprintf("f%d", memberIndex)
+				// f0 := []*string{}
+				out += varEmptyConstructorSDKType(
+					cfg, r,
+					memberVarName,
+					memberShape,
+					indentLevel+1,
+				)
+
+				//  f0 = append(f0, sourceVarName)
+				out += fmt.Sprintf("%s\t%s = append(%s, %s)\n", indent,
+					memberVarName, memberVarName, resVarPath)
+
+				// res.SetIds(f0)
+				out += setSDKForScalar(
+					cfg, r,
+					memberName,
+					targetVarName,
+					inputShape.Type,
+					sourceVarName,
+					memberVarName,
+					memberShapeRef,
+					indentLevel+1,
+				)
+			}
+		default:
+			// For rare ReadMany that have a singular identifier field.
+			// ex: DescribeReplicationGroups
```

Comment:
Not necessarily rare. :) Pretty much all of the RDS ReadMany APIs are like this.

Por ejemplo:

https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBParameterGroups.html
https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBParameters.html
https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBProxies.html
https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBProxyEndpoints.html

That last one is interesting because it has *two* scalar-typed filter fields :)

Also, there may be scalar-typed fields in a ReadMany input shape that act as filters on, say, boolean fields.

For example:

https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeOrderableDBInstanceOptions.html

Check out the `Vpc` member of the input shape. It's a boolean. Yep. This is why we can't have nice things.

Reply:
Woww thanks for these; I'll update the comment. Just when you think you were starting to get the hang of the API...